### PR TITLE
Include basic lexical matching pipeline to mondo ingest

### PR DESCRIPTION
**Pull request by matentzn**
- This is a skeleton setup to fix https://github.com/monarch-initiative/mondo-ingest/issues/38
 - All checkmarks in https://github.com/INCATools/ontology-access-kit/issues/191#event-7018102236 should be met.

### CODE REVIEW

**Comment by hrshdhgd** [comment]
Thoughts @matentzn ? As of now, the `metadata` and `curie_map` are injected outside of OAK (it's done within this project). The `lexmatch` feature does not have any provision to take a yaml input as of now (`metadata/mondo.sssom.config.yml`). Does OAK be held responsible to inject the `metadata` and `curie_map`?

cc:  @cmungall

File: src/ontology/mondo-ingest.Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@ report-mapping-annotations
 ### Standard constants #############
 ####################################
 MAPPINGSDIR=../mappings
-
+SCRIPTDIR=../scripts
```
[the remark this answers is not in the document]
Comment (hrshdhgd) [reply src/ontology/mondo-ingest.Makefile:12]: Changed

File: src/scripts/match-mondo-sources-all-lexical.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+
+if __name__ == '__main__':
+    input = None
+    config_file = None
+    output = None
+    rules = None
+    # TODO: there must be a better way to parse arguments.
+    for arg in sys.argv:
```
[the remark this answers is not in the document]
Comment (hrshdhgd) [reply src/scripts/match-mondo-sources-all-lexical.py]: Fixed!

File: src/scripts/match-mondo-sources-all-lexical.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    with open(config, "r") as f:
+        sssom_yaml = yaml.safe_load(f)
+        
+    resource = OntologyResource(slug=f"sqlite:///{Path(input).absolute()}")
+    oi = SqlImplementation(resource=resource)
+    lexical_index = create_lexical_index(oi)
+    save_lexical_index(lexical_index, OUT_INDEX_DB)
+    msdf = lexical_index_to_sssom(oi, lexical_index)
+    if rules:
+        msdf = lexical_index_to_sssom(oi, lexical_index, ruleset=load_mapping_rules(rules))
```
Comment (matentzn) [inline src/scripts/match-mondo-sources-all-lexical.py:80]: Matches in the sssom file you have checked in are still all `close`?

File: src/scripts/match-mondo-sources-all-lexical.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+)
+from sssom.writers import write_table
+import sys
+from pathlib import Path
+import yaml
+import click
+
+SRC = Path(__file__).resolve().parents[1]
+ONTOLOGY_DIR = SRC / "ontology"
+OUT_INDEX_DB = ONTOLOGY_DIR / "tmp/merged.db.lexical.yaml"
```
Comment (matentzn) [inline src/scripts/match-mondo-sources-all-lexical.py:31]: Is this way to construct paths platform independent? I.w. will it work on windows?

**Review by matentzn — COMMENTED** [review]
love it so far

File: src/scripts/match-mondo-sources-all-lexical.py (hunk 4 above)
Comment (hrshdhgd) [reply src/scripts/match-mondo-sources-all-lexical.py:31]: Yes, here's a [blog](https://realpython.com/python-pathlib/) that states it.

File: src/scripts/match-mondo-sources-all-lexical.py (hunk 3 above)
Comment (hrshdhgd) [reply src/scripts/match-mondo-sources-all-lexical.py:80]: I think it has something to do with the rules.yaml. If we don't provide any, the default for all would be `close` as per my [comment here](https://github.com/INCATools/ontology-access-kit/issues/98#issuecomment-1190725945).

Comment (hrshdhgd) [reply src/scripts/match-mondo-sources-all-lexical.py:80]: I confirmed it! It is the rules file that determines the matches in the file.

Comment (matentzn) [reply src/scripts/match-mondo-sources-all-lexical.py:80]: Yes, lets add a rules file permanentlt to `src/ontology/config/mondo-match-rules.yaml` and work on tweaking that moving forwards. Thank you for checking!

File: src/scripts/match-mondo-sources-all-lexical.py (hunk 4 above)
Comment (matentzn) [reply src/scripts/match-mondo-sources-all-lexical.py:31]: Thank you :)

File: src/ontology/config/mondo-match-rules.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
+rules:
```
Comment (matentzn) [inline src/ontology/config/mondo-match-rules.yaml:1]: looks great! Did you update teh table with that yet?

Comment (hrshdhgd) [reply src/ontology/config/mondo-match-rules.yaml:1]: Yes.

**Pull request merged**